float32 dim
---
float32[] x
float32[] y
